abcclassroom/feedback_report.py

```diff
@@ -0,0 +1,152 @@
+"""
+abc-classroom.feedback-report
+==============================
+
+"""
+
+import subprocess
+from pathlib import Path
+import csv
+
+import nbclean
+import nbformat as nf
+
+from . import config as cf
+
+
+def remove_notebook_test_cells(nb):
+    """
+    Clean the notebook's hidden cells using tags.
+
+    Parameters
+    ----------
+        nb: A string or notebook node as read in by nbformat.
+
+    Return
+    ------
+        The notebook with any cells tagged with 'hide'
+        removed.
+    """
+
+    cleaner = nbclean.NotebookCleaner(nb)
+    cleaner.clear(kind="content", tag="hide")
+
+    # Scrub all hidden tests from the notebook
+    text_replace_begin = "### BEGIN HIDDEN TESTS"
+    text_replace_end = "### END HIDDEN TESTS"
+    cleaner.replace_text(text_replace_begin, text_replace_end)
+
+    return cleaner.ntbk
+
+
+# Create a function that opens a notebook from a specific dir, cleans it
+# Saves the cleaned notebook to some place and moves the html to another place
+
+
+def create_clean_report(assignment_name):
+    """This function will take a notebook from a dir structure as follows:
+    dir/student-name/assignment-name/file-names.ipynb It will then open the
+    file,
+    clean and save the output as file-name-cleaned.ipynb.
+
+    Important: this assumes that we are cleaning jupyter notebooks. This will
+    not work on any other file format for feedback!!
+
+    For the time being I will save the cleaned notebook to a tmp dir that
+    will then be deleted once the html is created."""
+
+    # TODO: I think we do this a bunch so is it worth a helper for it
+    config = cf.get_config()
+    roster_filename = cf.get_config_option(config, "roster", True)
+    course_dir = cf.get_config_option(config, "course_directory", True)
+    materials_dir = cf.get_config_option(config, "course_materials", True)
+
+    # This code was taken from the feedback.py module. It should become a
+    # helper potentially if we end up using it.
+    try:
+        # This assumes the feedback directory exists. but we should test that
+        # it exists
+        autograded_dir = Path(course_dir, materials_dir, "autograded")
+        with open(roster_filename, newline="") as csvfile:
+            reader = csv.DictReader(csvfile)
+            for row in reader:
+                student = row["github_username"]
+
+                # Create path to autograded directory & get all notebooks
+                path_to_notebooks = Path(
+                    autograded_dir, student, assignment_name
+                )
+                graded_notebooks = path_to_notebooks.glob("*.ipynb")
```

Comment:
Then add another try / except here in case the student autograded dir doesn't exist. 
If it doesnt we want to just keep going and do as many as we can. 

Breaking up try/ excepts will allow us to provide more specific error reporting to the user